staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -83,3 +95,30 @@ func (record *attributesRecord) GetOldObject() runtime.Object {
 func (record *attributesRecord) GetUserInfo() user.Info {
 	return record.userInfo
 }
+
+func (record *attributesRecord) SetAnnotations(pluginName string, annotations map[string]string) {
+	if record.annotations == nil {
```

Comment:
skipping on nil annotations was unexpected... why wouldn't we initialize to an empty map here?

Reply:
1. This makes annotations inconsistent with other elements.
    For other elements, their reference can be hold by the outside world of Attribute.

2. If we  initialize to an empty map here, we need a new interface GetAnnotations(). And we need to remind users of this interface, returned annotations map is not goroutine safe.